### ISSUE TITLE
Saving FPU registers during IRQ Handler 

### ISSUE THREAD

**Issue by Surysingh** [littlekernel/lk#385]
Hi All,  
For Aarch64,  in file arch\arm64\exceptions.s,  for arm64_irq_current_el_SPx, why floating point state and registers are not saved?

Is there not a chance that interrupt service routine /  handler will use and hence corrupt the floating point registers

**Comment by travisg** [comment]
This is correct. Nothing in the IRQ or scheduler path should use the FPU, so the context is not saved. This speeds things up greatly, since it takes quite a bit of time and space to save the full fpu/vector state.
